service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -537,51 +551,111 @@ private ApiCloneControlledGcpBigQueryDatasetResult fetchCloneBigQueryDatasetResu
-    // when the resource is public
+  private PrivateUserRole validatePrivateResourceUser(
+      UUID workspaceId,
+      ApiControlledResourceCommonFields commonFields,
+      AuthenticatedUserRequest userRequest,
+      ControlledResourceIamRole defaultUserRole) {
+
     AccessScopeType accessScope = AccessScopeType.fromApi(commonFields.getAccessScope());
-    if (accessScope == AccessScopeType.ACCESS_SCOPE_PRIVATE && privateRoles.isEmpty()) {
-      throw new ValidationException("At least one IAM role is required for private resources");
+    ManagedByType managedBy = ManagedByType.fromApi(commonFields.getManagedBy());
+    ApiPrivateResourceUser inputUser = commonFields.getPrivateResourceUser();
+
+    // Shared access has no private user role
+    if (accessScope == AccessScopeType.ACCESS_SCOPE_SHARED) {
+      validateNoInputUser(inputUser);
+      return new PrivateUserRole(false, null, null);
+    }
+
+    // Private access scope
+    switch (managedBy) {
+      case MANAGED_BY_APPLICATION:
+        {
+          // Supplying a user is optional for applications
+          if (inputUser == null) {
+            return new PrivateUserRole(false, null, null);
+          }
+
+          // We have a private user, so make sure the email is present and valid
+          String userEmail = commonFields.getPrivateResourceUser().getUserName();
+          ControllerValidationUtils.validateEmail(userEmail);
+
+          /*
+           TODO: the APPLICATION role does not have read_policies action, so cannot perform
+            this check. We need to give it read_policies.
+          // Validate that the assigned user is a member of the workspace. It must have at least READ action.
+           SamRethrow.onInterrupted(
+               () -> samService.userIsAuthorized(
+                   SamConstants.SAM_WORKSPACE_RESOURCE,
+                   workspaceId.toString(),
+                   SamConstants.SAM_WORKSPACE_READ_ACTION,
+                   userEmail,
+                   userRequest), "validate private user is workspace member");
+           */
+          // Translate the incoming role list into our internal model form
+          // This also validates that the incoming API model values are correct.
+          List<ControlledResourceIamRole> roles =
+              commonFields.getPrivateResourceUser().getPrivateResourceIamRoles().stream()
+                  .map(ControlledResourceIamRole::fromApiModel)
+                  .collect(Collectors.toList());
+          if (roles.isEmpty()) {
+            throw new ValidationException(
+                "You must specify at least one role when you specify PrivateResourceIamRoles");
+          }
+
+          // The legal options for the assigned user of an application is READER
+          // or WRITER. EDITOR is not allowed. We take the "max" of READER and WRITER.
+          var maxRole = ControlledResourceIamRole.READER;
+          for (ControlledResourceIamRole role : roles) {
```

Comment:
Maybe since I already inadvertently changed the REST API, I should get rid of the list and only allow one role. I was keeping the list for compatibility sake.

Reply:
I put off the REST API change until CLI and UI can be updates. I'll leave this as is for the time being.